packages/manager/apps/dedicated/client/app/ip/byoip/byoip.html

```diff
@@ -0,0 +1,158 @@
+            data-description="{{:: 'ip_byoip_region_description' | translate}}"
+            data-navigation="$ctrl.byoip.campus"
+            data-prevent-next="true"
+        >
+            <div class="row mb-3">
+                <oui-select-picker
+                    disabled="$ctrl.byoip.campus && $ctrl.regions.length === 1"
+                    data-ng-repeat="region in $ctrl.regions track by $index"
+                    class="col-md-4 mb-3"
+                    data-model="$ctrl.byoip.campus"
+                    values="[region]"
+                    data-label="{{:: 'ip_byoip_region_' + region | translate }}"
+                    data-variant="light"
+                    data-on-change="$ctrl.onRegionSelect(modelValue)"
+                >
+                    <oui-select-picker-section class="text-uppercase"
+                        ><span data-ng-bind="region"></span
+                    ></oui-select-picker-section>
+                </oui-select-picker>
+            </div>
+            <div class="row mb-3">
+                <h3 data-translate="ip_byoip_region_token_title"></h3>
+                <p data-translate="ip_byoip_region_token_description"></p>
+                <p data-translate="ip_byoip_region_token_description_1"></p>
+                <p
+                    data-translate="ip_byoip_region_token_description_2"
+                    class="mb-3"
+                ></p>
+
+                <oui-clipboard
+                    data-model="$ctrl.token"
+                    class="col-md-6"
+                ></oui-clipboard>
+            </div>
+        </oui-step-form>
+        <!-- IP range -->
+        <oui-step-form
+            data-header="{{:: 'ip_byoip_select_ip_range_title' | translate }}"
+            data-submit-text="{{:: 'ip_byoip_submit_text' | translate }}"
+            data-valid="$ctrl.byoip.ip"
+            data-description="{{:: 'ip_byoip_select_ip_range_description' | translate}}"
+            data-navigation="$ctrl.byoip.ip"
+            data-prevent-next="true"
+        >
+            <div class="row">
+                <oui-field
+                    class="col-md-6"
+                    label="{{::'ip_byoip_select_ip_range_input_label' | translate}}"
+                >
+                    <input
+                        class="oui-input"
+                        name="field"
+                        placeholder="192.168.22.42/24"
+                        data-type="text"
+                        data-ng-pattern="$ctrl.IPV4_BLOCK_PATTERN"
+                        data-ng-model="$ctrl.byoip.ip"
+                        data-required
+                    />
+                </oui-field>
+            </div>
+        </oui-step-form>
+        <!-- Choose autonomous system -->
+        <oui-step-form
+            data-header="{{:: 'ip_byoip_as_system_title' | translate }}"
+            data-submit-text="{{:: 'ip_byoip_submit_text' | translate }}"
+            data-valid="$ctrl.asType"
```

Comment:
Don't you have to handle condition on `$ctrl.byoip.asRir` and `$ctrl.byoip.asNumber` here to validate the step?

Reply:
It has to work for 2 conditions here: 

1. selected asType is `ovh_cloud` : can submit directly
2. selected `own` : have to add `arin/ripe` value along with `asNumber`

Since i have required on these fields, validation is working properly for both conditions